### PR TITLE
:recycle: refactor: Migrate HealthChecker to v3

### PR DESCRIPTION
**Pull request by luk3skyw4lker**
## Description

Related to #2863 

## Changes Introduced

Migrating the healthchecker middleware to v3, using the proposition made by @ReneWerner87 in the above mentioned PR. Here is the benchmark and successful test results:
image

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (improvement to existing features and functionality)
- [ ] Documentation update (changes to documentation)
- [ ] Performance improvement (non-breaking change which improves efficiency)
- [ ] Code consistency (non-breaking change which improves code reliability and robustness)

## Checklist

Before you submit your pull request, please make sure you meet these requirements:

- [x] Followed the inspiration of the Express.js framework for new functionalities, making them similar in usage.
- [x] Conducted a self-review of the code and provided comments for complex or critical parts.
- [x] Updated the documentation in the `/docs/` directory for [Fiber's documentation](https://docs.gofiber.io/).
- [x] Added or updated unit tests to validate the effectiveness of the changes or new features.
- [x] Ensured that new and existing unit tests pass locally with the changes.
- [x] Verified that any new dependencies are essential and have been agreed upon by the maintainers/community.
- [x] Aimed for optimal performance with minimal allocations in the new code.
- [x] Provided benchmarks for the new code to analyze and improve upon.

## Summary by CodeRabbit

- **New Features**
	- Enhanced customization for health check middleware in Fiber apps, allowing for custom probes for liveness and readiness checks.
- **Refactor**
	- Simplified the configuration structure for health checks by consolidating probe functions.
	- Unified the approach for handling liveness and readiness checks.
- **Documentation**
	- Updated documentation to reflect new customization capabilities and configuration simplifications.
- **Tests**
	- Added new routes and updated tests to align with the refactored health check functionality.

### CODE REVIEW

**Comment by luk3skyw4lker** [comment]
@ReneWerner87 @gaby I don't know how to explain this, but after [this commit](https://github.com/gofiber/fiber/commit/4a1891b4b94824ee0f0975cddb39475d77d4a97d) at the v2ToV3Merge branch, the performance of the middleware dropped in half:
image

Although the commit did not changed anything in the middleware itself, the performance dropped drastically, at least running locally. I'll push the changes to see what's the results

**Comment by luk3skyw4lker** [comment]
Seems like it wasn't only healthchecker that suffered from this:
image

**Comment by gaby** [comment]
> Seems like it wasn't only healthchecker that suffered from this: image

Thats fine, we can address that in #2864 I'm more concern with the tests failing

**Comment by gaby** [comment]
Performance Diff between Sequential / Parallel:

```
Benchmark_HealthCheck-4            	 6781260	       179.2 ns/op	       0 B/op	       0 allocs/op
Benchmark_HealthCheck_Parallel-4   	14886872	        85.98 ns/op	       0 B/op	       0 allocs/op
```

**Comment by ReneWerner87** [comment]
> @ReneWerner87 @gaby I don't know how to explain this, but after [this commit](https://github.com/gofiber/fiber/commit/4a1891b4b94824ee0f0975cddb39475d77d4a97d) at the v2ToV3Merge branch, the performance of the middleware dropped in half: image
> 
> Although the commit did not changed anything in the middleware itself, the performance dropped drastically, at least running locally. I'll push the changes to see what's the results

what was the performance in the v2 branch ? 

its maybe related to my fix https://github.com/gofiber/fiber/pull/2864/commits/5a0167a460ef2385ecade84683bd0007a28b97c1
for this real bug https://github.com/gofiber/fiber/issues/2871#issuecomment-1958146253

**Comment by luk3skyw4lker** [comment]
@ReneWerner87 It was around 120 ns/op according to this comment: https://github.com/gofiber/fiber/pull/2863#issuecomment-1951475681

But the performance before the commit and with the migration was around 110 ns/op according to this PR. I'll try and run it again with @gaby commits

**Comment by luk3skyw4lker** [comment]
```
goos: linux
goarch: amd64
pkg: github.com/gofiber/fiber/v3/middleware/healthcheck
cpu: Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz
Benchmark_HealthCheck-4                  5853678               198.3 ns/op             0 B/op          0 allocs/op
Benchmark_HealthCheck-4                  5897978               199.9 ns/op             0 B/op          0 allocs/op
Benchmark_HealthCheck-4                  5854675               197.6 ns/op             0 B/op          0 allocs/op
Benchmark_HealthCheck-4                  5786534               197.8 ns/op             0 B/op          0 allocs/op
Benchmark_HealthCheck-4                  5776459               199.2 ns/op             0 B/op          0 allocs/op
Benchmark_HealthCheck_Parallel-4        17670471                58.34 ns/op            0 B/op          0 allocs/op
Benchmark_HealthCheck_Parallel-4        18804096                64.86 ns/op            0 B/op          0 allocs/op
Benchmark_HealthCheck_Parallel-4        19699030                63.55 ns/op            0 B/op          0 allocs/op
Benchmark_HealthCheck_Parallel-4        20749539                75.82 ns/op            0 B/op          0 allocs/op
Benchmark_HealthCheck_Parallel-4        17202957                74.76 ns/op            0 B/op          0 allocs/op
PASS
ok      github.com/gofiber/fiber/v3/middleware/healthcheck      16.499s
```

Looks the same as before to me, but since @gaby said it can be addressed in #2864 which also has other affected benchmarks

**Comment by luk3skyw4lker** [comment]
@gaby yep, noticed that too, I'm gonna be taking care of some code rabbit comments right now and the PR should be good to go, do you guys have any other ideas for changes/improvements?

File: middleware/healthcheck/healthcheck_test.go [hunk 1]
```diff
@@ -123,22 +65,21 @@ func Test_HealthCheck_Custom(t *testing.T) {
...
-		LivenessProbe: func(_ fiber.Ctx) bool {
+	app.Get("/live", NewHealthChecker(Config{
+		Probe: func(_ fiber.Ctx) bool {
 			return true
 		},
-		LivenessEndpoint: "/live",
-		ReadinessProbe: func(_ fiber.Ctx) bool {
+	}))
+	app.Get("/ready", NewHealthChecker(Config{
+		Probe: func(_ fiber.Ctx) bool {
```
[the remark this answers is not in the document]
Comment (luk3skyw4lker) [reply middleware/healthcheck/healthcheck_test.go:74]: The 503 case is already covered in line 130 and 132.

**Review by gaby — APPROVED** [review]
LGTM

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**